### PR TITLE
Document Target_Update [Target Docs 4/?]

### PR DESCRIPTION
**Pull request by fig02**
This documents `Target_Update` and every `TargetContext` variable.
The old PR was referenced for inspiration but many names and comments are new.

I'd read `Target_Update` and `TargetContext` top to bottom, then everything else which are rename side effects

### CODE REVIEW

File: src/code/z_actor.c [hunk 1]
```diff
@@ -447,101 +449,128 @@ void Target_Draw(TargetContext* targetCtx, PlayState* play) {
...
+            targetCtx->reticleActor = curLockOnActor;
 
-            if (actorArg->id == ACTOR_EN_BOOM) {
+            if (curLockOnActor->id == ACTOR_EN_BOOM) {
+                // Don't draw the reticle when locked onto the boomerang.
+                // Note that it isn't possible to lock onto the boomerang, so this code doesn't do anything.
+                // This implies that the boomerang camera lock may have been implemented with Z-Targeting at one point,
+                // but was eventually implemented as its own camera mode instead.
                 targetCtx->reticleFadeAlphaControl = 0;
             }
```
Comment (fig02) [inline src/code/z_actor.c:541]: For anyone curious on my claims here, I removed `CAM_MODE_FOLLOW_BOOMERANG` camera mode and made the boomerang manually targetable to make this code actually able to manifest:

https://github.com/user-attachments/assets/c36899c7-afd5-47b1-a0f0-c5fa2f784c3b

File: src/code/z_actor.c [hunk 2]
```diff
@@ -294,34 +294,36 @@ void Target_InitReticle(TargetContext* targetCtx, s32 actorCategory, PlayState*
...
 
     Target_SetNaviState(targetCtx, actor, actor->category, play);
     Target_InitReticle(targetCtx, actor->category, play);
 }
 
 void Target_Draw(TargetContext* targetCtx, PlayState* play) {
-    Actor* actor = targetCtx->lockOnActor;
+    Actor* actor; // used for both the reticle and arrow
+
+    actor = targetCtx->reticleActor;
```
Comment (krm01) [inline src/code/z_actor.c:326]: it seems like PRs recently have been lifting assignments up into the declaration when they can, why have you chosen to go the other way here and undo it?

File: src/code/z_actor.c [hunk 3]
```diff
@@ -447,101 +449,128 @@ void Target_Draw(TargetContext* targetCtx, PlayState* play) {
...
+        targetCtx->reticlePos.x = curLockOnActor->world.pos.x;
+        targetCtx->reticlePos.y =
+            curLockOnActor->world.pos.y - (curLockOnActor->shape.yOffset * curLockOnActor->scale.y);
+        targetCtx->reticlePos.z = curLockOnActor->world.pos.z;
 
         if (targetCtx->reticleSpinCounter == 0) {
-            f32 temp5 = (500.0f - targetCtx->reticleRadius) * 3.0f;
-            f32 temp6 = (temp5 < 30.0f) ? 30.0f : ((100.0f < temp5) ? 100.0f : temp5);
+            f32 step = (500.0f - targetCtx->reticleRadius) * 3.0f;
+            f32 reticleZoomStep = (step < 30.0f) ? 30.0f : ((100.0f < step) ? 100.0f : step);
```
Comment (krm01) [inline src/code/z_actor.c:557]: does the `CLAMP` macro match here?

File: src/code/z_actor.c (hunk 2 above)
Comment (fig02) [reply src/code/z_actor.c:326]: basically to raise importance to the fact that it is a multi use variable, and give room for my comment on that fact. I want to say "here is the first assignment to this variable" not "this is the only assignment to this variable"

File: src/code/z_actor.c (hunk 3 above)
Comment (fig02) [reply src/code/z_actor.c:557]: I believe it was clamp at some point in the past but it got removed, I suppose to match retail. I can try tho

Comment (fig02) [reply src/code/z_actor.c]: nvm it seems fine

**Review by krm01 — APPROVED** [review]

**Review by cadmic — APPROVED** [review]

**Pull request merged**